boards/xtensa/esp32/esp32-lyrat/include/board_memorymap.h

```diff
@@ -0,0 +1,128 @@
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+#ifndef __BOARDS_XTENSA_ESP32_ESP32_LYRAT_INCLUDE_BOARD_MEMORYMAP_H
+#define __BOARDS_XTENSA_ESP32_ESP32_LYRAT_INCLUDE_BOARD_MEMORYMAP_H
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <stdint.h>
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+/* Kernel ROM */
+
+#define KIROM_START     (uintptr_t)&__kirom_start
+#define KIROM_SIZE      (uintptr_t)&__kirom_size
+#define KDROM_START     (uintptr_t)&__kdrom_start
+#define KDROM_SIZE      (uintptr_t)&__kdrom_size
+
+/* Kernel RAM */
+
+#define KIRAM_0_START   (uintptr_t)&__kiram_0_start
+#define KIRAM_0_SIZE    (uintptr_t)&__kiram_0_size
+#define KIRAM_0_END     (uintptr_t)&__kiram_0_end
+#define KIRAM_1_START   (uintptr_t)&__kiram_1_start
+#define KIRAM_1_SIZE    (uintptr_t)&__kiram_1_size
+#define KIRAM_1_END     (uintptr_t)&__kiram_1_end
+#define KDRAM_0_START   (uintptr_t)&__kdram_0_start
+#define KDRAM_0_SIZE    (uintptr_t)&__kdram_0_size
+#define KDRAM_0_END     (uintptr_t)&__kdram_0_end
+#define KDRAM_1_START   (uintptr_t)&__kdram_1_start
+#define KDRAM_1_SIZE    (uintptr_t)&__kdram_1_size
+#define KDRAM_1_END     (uintptr_t)&__kdram_1_end
+
+/* Exception vectors */
+
+#define VECTORS_START   (uintptr_t)&__vectors_start
+#define VECTORS_END     (uintptr_t)&__vectors_end
+
+/* User ROM */
+
+#define UIROM_START     (uintptr_t)&__uirom_start
+#define UIROM_SIZE      (uintptr_t)&__uirom_size
+#define UIROM_END       (uintptr_t)&__uirom_end
+#define UDROM_START     (uintptr_t)&__udrom_start
+#define UDROM_SIZE      (uintptr_t)&__udrom_size
+#define UDROM_END       (uintptr_t)&__udrom_end
+
+/* User RAM */
+
+#define UIRAM_START     (uintptr_t)&__uiram_start
+#define UIRAM_SIZE      (uintptr_t)&__uiram_size
+#define UIRAM_END       (uintptr_t)&__uiram_end
+#define UDRAM_START     (uintptr_t)&__udram_start
+#define UDRAM_SIZE      (uintptr_t)&__udram_size
+#define UDRAM_END       (uintptr_t)&__udram_end
```

Comment:
let's follow this change:
https://github.com/apache/incubator-nuttx/pull/7160/files#diff-519d1b034dded84d8087346fc7191e18e93f5c1f592a55f85b465349bec84225L36-L127